### ISSUE TITLE
<listBibl> isn't allowed as a child of <back>

### ISSUE THREAD

**Issue by TEITechnicalCouncil** [TEIC/TEI#1169]
The Element &lt;listBibl&gt; is not allowed as child of &lt;back&gt;. This enforces to wrap it with a &lt;div&gt; or something else in the backmatter. That makes no sense to me.

Original comment by: olafxberg

**Comment by ebeshero** [comment]
Per Daniel Schopper's message to the TEI list, "the fix misses `<listBibl>` since it is not part of model.listLike" (so that it is still not permitted as a child of `<back>`).

**Comment by ebeshero** [comment]
I'm happy to take on the repair, unless there's any discussion. Any reason listBibl should not be a member of listLike?

**Comment by sydb** [comment]
I don’t know, but I suspect there is a reason `<listbibl>` is not in model.listLike; and putting it into model.frontPart may make more sense.

**Comment by jamescummings** [comment]
The reason is likely to be content-model based -- model.listLike and listBibl are probably independently defined in various content models and thus you'llg et ambiguous content models. See, for just one example: sourceDesc.  This has model.biblLike (which contains listBibl) and model.listLike (which contains the other lists).  All of these could reasonably be in sourceDesc (some more than others).  

So if you just have listBibl claim membership model.listLike then your DTDs will be ruined. (Of course, we could just stop supporting DTDs...)

**Comment by ebeshero** [comment]
It seems odd to put listBibl in model.frontPart, since often a bibliography is associated specifically with back matter.

**Comment by ebeshero** [comment]
Would it be more reasonable to include `<back>` in model.bibLike?

**Comment by sydb** [comment]
I think you mean to put model.blblLike into the content of `<back>`, no?
It might be. But my instinct is to look carefully at the effects of putting `<listBibl>` into either model.listLike or model.frontPart first. (BTW, the content models of `<front>` and `<back>` are pretty much exactly the same; it could just as well have been called “model.front_or_back_Part” :-)

**Comment by ebeshero** [comment]
Oops--right, that's what I meant: to suggest putting model.biblLike in the content of `<back>`, though now that I look at it, we really don't seem to have included bibLike in the main structural divisions of a TEI document: http://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-model.biblLike.html  

I hope it can be in listLike *and* frontPart.

**Comment by lb42** [comment]
As James pointed out, the problem is that listBibl is a member of model.biblLike. Introducing it anywhere where model.listLike is also permitted itherefore nevitably introduces an ambiguity. 
A work around is to wrap your listBibl inside a &lt;div> (yuk) or a &lt;ab> or a &lt;p> or even a &lt;list>&lt;itme> of coiurse. But the only simple solutions I can see are
 - addt &lt;listBibl explicitly to the content model for &lt;back>  (and also for &lt;front>; see note on the latter
  - remove &lt;listBibl from model.biblLike and handle the resulting chaos

**Comment by sydb** [comment]
Worth noting that there is a comment in the source tagdoc for `<listBibl>` that says:
~~~~~xml
    <!--memberOf key="model.listLike"/-->
    <!-- incompatible with membership in model.biblLike since both are
referenced from macro.inter -->
~~~~~
So I figure we can probably cross that idea off. But just to be sure, I locally built a version of P5 with `<listBibl>` in model.listLike, and there were almost 100 errors just from the tests inside the build process, all attributable to that one change.

I also generated a version of P5 with `<listBibl>` in model.frontPart, and there were no extra errors during the build process, and the tei_all.xsd file it generated was valid. The tei_all.dtd file it generated is not quite valid, but the errors oXygen flags have nothing to do with `<listBibl>`. So I’m leaning towards this as the way to go.

**Comment by ebeshero** [comment]
Wow! Sounds like someone tested the model for listBibl before us...and this was nowhere near as straightforward a fix as I thought. Should we proceed with adding it to frontPart, or is it better to leave it alone--any more chances of unleashing chaos?

**Comment by sydb** [comment]
I think the original request (to allow `<listBibl>` as a child of `<back>`, and therefore of `<front>`) is quite a reasonable one. I think we should definitely do that. The question is how. The model.frontPart class occurs nowhere else but in `<front>` and `<back>`, so adding it there seems quite safe. Unless there are objections, that is what I’ll plan to do.

**Comment by ebeshero** [comment]
No objections here...anyone else?

**Comment by lb42** [comment]
Easy enough to test this approach by writing an ODD! If there are problems in DTD generation, they should be fixed though.

**Comment by sydb** [comment]
And (as mentioned above), I’ve also tested it with a complete build of P5, and that worked. So I’m just waiting around a bit longer to see if anyone raises any objections, and then will check in the change. (I.e., will put `<listBibl>` into model.frontPart.) I will post a prod to TEI Council list soon.

**Comment by jamescummings** [comment]
Seems reasonable to me.  Though I think we should also sort out the whole model.biblLike and model.listLike pile of spaghetti sometime.

**Comment by sydb** [comment]
Fixed in a17a766; will close ticket as soon as I see Mr. Jenkins is happy.
